src/Microsoft.Health.Fhir.Api/Features/Health/IDatabaseStatusReporter.cs

```diff
@@ -0,0 +1,25 @@
+// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Diagnostics.HealthChecks;
+using Microsoft.Health.Encryption.Customer.Health;
+
+namespace Microsoft.Health.Fhir.Api.Features.Health
+{
+    /// <summary>
+    /// Provides functionality to report the status of Data Store.
+    /// </summary>
+    public interface IDatabaseStatusReporter
+    {
+        /// <summary>
+        /// Gets the current status of the customer-managed key.
+        /// </summary>
+        /// <param name="cancellationToken">A token to cancel the operation.</param>
+        /// <returns>A task that returns true if the key is healthy; otherwise, false.</returns>
+        Task<HealthCheckResult> IsCustomerManagerKeyProperlySetAsync(CancellationToken cancellationToken);
```

Comment:
I believe this method should return a boolean instead of a HealthCheckResult. 

The reason for this changes it's separation of responsibilities. 

When calling method _IsCustomerManagerKeyProperlySetAsync_ you should be a result true or false. 
The execution of the method should be agnostic to Health Check (as the returned type if called Health Check Result). 

Additionally to that, the class responsible for creating HealthCheckResults should be StorageInitializedHealthCheck.